src/vkscript/command_parser_test.cc

```diff
@@ -3339,7 +3339,7 @@ TEST_F(CommandParserTest, Uniform) {
   const auto* buf = cmd->GetBuffer();
   const auto* values = buf->GetValues<float>();
   std::vector<float> results = {2.1f, 3.2f, 4.3f, 0.f};
-  ASSERT_EQ(results.size(), buf->ValueCount());
+  EXPECT_TRUE(results.size() < buf->ValueCount());
```

Comment:
This shouldn't be needed? Something has gone wrong if we have more then 4 values.

Reply:
buf->ValueCount()
= element_count_ * format_->ValuesPerElement()
= element_count_ * ValuesPerRow() * column_count_ && (is_std140_ && column_count_ > 1)
= element_count_ * 4 * column_count_
= element_count_ * 4 * 1
= element_count_ * 4
As a result:
with my change in command_parser.cc: element_count_ == 3 -> buf->ValueCount() == 12
without it: element_count_ == 1 -> buf->ValueCount() == 4

I'm a little lost! which element_count is correct for a "uniform vec3 32 2.1 3.2 4.3"?

Reply:
So, the distinction between element and value. A vec3 is 1 element, but it's 4 values (4, not 3 due to the layout rules padding the last item). So, a `vec3 32 2.1 3.2 4.3` is a single element with 4 values.

The `element count * format->ValuesPerElement()` should give the correct number of values.

I'd actually expect that to return `12` instead of `9`. The offset of 32 bytes of float's means we have 2 elements, or 8 values in the offset, plus 4 values that are being appended.

Reply:
I'm finding these functions a little confusing. Let's go back to the initial problem: Dawn need the  size to create a buffer. 
Do you have any suggestion about how to set it when there a ubo command in a vkscript?

Reply:
So, it sounds like this is going to get tricky with VkScript and Dawn. In VkScript, the buffers can grow as the program executes. With AmberScript the buffers are static in size. There is no way to adjust the size of a buffer in Dawn? Can we, as the buffers change size delete the old ones and create new ones?

Reply:
Dawn CreateBuffer API needs the size as an argument. Later on we can do SetSubData however it shouldn't over for the initial given size.
There is not DeleteBuffer and since we bind buffers to the pipeline we cannot just abandon the previous one and create a new one each time.

Reply:
Is this a current Dawn limitation they're going to fix, or are buffers just immutable in size in Dawn and that's the way it will always be?

I guess currently we'll have to make sure the buffer is always resized to the largest size needed. That being said, for this test, the size should be known to be exact, we shouldn't need to make the comparison weaker.

Reply:
Had a thought on the train this morning. When we're parsing the VkScript file should we maintain 2 sizes in the Buffer class. The `SizeInBytes` would be the initial size to make the buffer and a `MaxSizeInBytes` would be the total size needed to hold the buffer over all `ssbo size` and `ssbo subdata` size calls?

Then, the Vulkan side stays the same and uses the size and resizes as needed and the Dawn side can use the max size to set the buffer?

The buffer would initially be sized to whatever `SizeInBytes` comes out to be.

Reply:
taking a look